src/client_backend/triton_c_api/triton_loader.cc

```diff
@@ -1363,8 +1363,11 @@ TritonLoader::GetSingleton()
 
 TritonLoader::~TritonLoader()
 {
-  TRITONSERVER_Error* allocator_err = response_allocator_delete_fn_(allocator_);
```

Comment:
Calling Delete() from main causes the static instance to be instantiated, which means at end of process, destructor gets called here, but if Triton C API mode wasn't used, none of these function pointers is initialized, so it caused a set fault. Fix was to guard against uninitialized function pointer.

Reply:
Right, I see, yep. I hadn't spotted this one before. There's definitely some overall hardening we should be doing here, with better RAII-style usage, but this totally will do for now.